### DISCUSSION TITLE
Passwordless social login with OAuth2.0

### DISCUSSION THREAD

**Discussion by limjiayi** [maxcountryman/axum-login#41] [Q&A] [answered]
I'm trying to implement signing in with Google using OAuth, and I also want to have role-based route protection.

To implement `AuthUser`, I must provide an id, a password hash and a role, but to me the whole point of using social auth is so that my users don't have to create yet another password and I don't have to deal with users forgetting and needing to reset passwords.

Can I replace the password hash with some other value?

**Comment by maxcountryman** [comment, marked as answer]
Yes you absolutely can. It should be a value that will change when the user updates their credentials ideally (wherever those might be managed).

**Reply by limjiayi** [reply]
Thanks! Since I'm using Sign In with Google and don't have the user's password for their Google Account, I guess I will try replacing the password hash with the email address the user has associated with their Google Account. If that changes, their session is invalidated.
